share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml

```diff
@@ -13,6 +13,7 @@ spack:
     - default_specs:
       - lz4  # MakefilePackage
       - mpich~fortran  # AutotoolsPackage
+      - tut  # WafPackage
```

Comment:
Why is this needed?

Every other change looks fine to me. I'd ask in general though:

* Why deprecate things that depend on python@2 (i.e. why not let the deprecation of Python 2 address that when it comes to it)?
* Do we need different deprecation "modes" i.e. something finer-grained than boolean:
  * Installing the package is allowed but warn
  * Installing the package is not allowed but existing installs are ok
  * Existing installs are not allowed

Reply:
> Why is this needed?

@tgamblin asked me to revert #28360, the WafPackage test was removed because tut became deprecated, but now that tut is no longer deprecated we should test again.

> Why deprecate things that depend on python@2 (i.e. why not let the deprecation of Python 2 address that when it comes to it)?

I chose to deprecate other packages in addition to EOL Python in #28003 so that we could get an idea of how many packages would be affected, and so that we could notify the maintainers of those packages. Now that EOL Python is no longer deprecated, we are also removing those package deprecations.

> Do we need different deprecation "modes" i.e. something finer-grained than boolean

@haampie suggested the same thing on Slack today. Regardless of whether or not we should do this, this doesn't affect Python packages. @varioustoxins (and presumably others) wants to be able to install Python 2-only packages using Spack. I'm assuming this also includes installing on systems without a Python 2 external. We need to remove deprecations from all packages regardless.

Reply:
> Do we need different deprecation "modes" i.e. something finer-grained than boolean

For reference, we discussed that issue when we hooked the `deprecated` keyword to the concretizer https://github.com/spack/spack/pull/23491#pullrequestreview-655611931